### PR TITLE
SG-955 - On domain verification error or failure, set last checked da…

### PR DESCRIPTION
**Pull request by JaredSnider-Bitwarden**
…te on the org domain.

## Type of change

```
- [X] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

During verification, update the org domain model's last checked date on failure or error. 

## Code changes

* **file.ext:** Description of what was changed and why

## Before you submit

- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- If making database changes - make sure you also update Entity Framework queries and/or migrations
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

**Review by gbubemismith — APPROVED** [review]

**Review by gbubemismith — COMMENTED** [review]
Nice job @JaredSnider-Bitwarden, I think the unit test for this would fail because of the latest changes

**Review by gbubemismith — COMMENTED** [review]
Lines 59, 60 and 61 are no longer needed

File: src/Core/OrganizationFeatures/OrganizationDomains/VerifyOrganizationDomainCommand.cs [hunk 1]
```diff
@@ -61,6 +66,9 @@ public async Task<OrganizationDomain> VerifyOrganizationDomain(Guid id)
             _logger.LogError("Error verifying Organization domain.", e);
         }
 
+        domain.SetLastCheckedDate();
+        await _organizationDomainRepository.ReplaceAsync(domain);
+
         await _eventService.LogOrganizationDomainEventAsync(domain, EventType.OrganizationDomain_NotVerified);
```
Comment (gbubemismith) [inline src/Core/OrganizationFeatures/OrganizationDomains/VerifyOrganizationDomainCommand.cs:72]: The LogOrganizationDomainEventAsyn can be updated with this

```
await _eventService.LogOrganizationDomainEventAsync(domain,
            orgDomain.VerifiedDate != null ? EventType.OrganizationDomain_Verified : EventType.OrganizationDomain_NotVerified);
```

Comment (JaredSnider-Bitwarden) [reply src/Core/OrganizationFeatures/OrganizationDomains/VerifyOrganizationDomainCommand.cs:72]: Great catch!

**Comment by JaredSnider-Bitwarden** [comment]
> Unit test needs to be updated to reflect latest change
I haven't had time to circle back to the required test updates.

**Pull request merged**